deepchem/models/torch_models/ferminet.py

```diff
@@ -124,46 +164,77 @@ def prepare_input_stream(self,) -> Tuple[Any, Any, Any, Any]:
                         self.electron_no[int(electro_neg[-1 - iter][0])][0] += 1
 
         total_electrons = np.sum(self.electron_no)
-        self.up_spin = (total_electrons + 2 * self.spin) // 2
-        self.down_spin = (total_electrons - 2 * self.spin) // 2
+
```

Comment:
Could you give me a summary of the refactoring changes you've made here?

Reply:
The changes I have done are:
1. added a new helper function prepare_hf which gives hartree fock orbitals values for pretraining.
2. Deleted the code to prepare input streams as it should be done in forwards, as we need the gradient of output wrt to the electron's coordinates we are going to pass

For now, this model is empty and have added the helper function only. In successive PRs I will complete the forward function and add  example